### PR TITLE
feat(rtc_auto_mode_manager)!: add auto mode status array

### PR DESCRIPTION
**Pull request by taikitanaka3**
## Description

This PR needs https://github.com/tier4/tier4_autoware_msgs/pull/70

add auto mode status array
Screenshot from 2022-12-18 12-37-51

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by taikitanaka3** [comment]
@tkimura4 @tkhmy 
can you review this PR please?

**Review by tkimura4 — APPROVED** [review]
LGTM

**Comment by taikitanaka3** [comment]
@rej55 could you approve this?

File: planning/rtc_auto_mode_manager/include/rtc_auto_mode_manager/node.hpp [hunk 1]
```diff
@@ -18,18 +18,24 @@
...
 {
+using tier4_rtc_msgs::msg::AutoModeStatusArray;
 class RTCAutoModeManagerNode : public rclcpp::Node
 {
 public:
   explicit RTCAutoModeManagerNode(const rclcpp::NodeOptions & node_options);
 
 private:
+  rclcpp::TimerBase::SharedPtr timer_;
+  AutoModeStatusArray auto_mode_statuses_;
```
[the remark this answers is not in the document]
Comment (taikitanaka3) [reply planning/rtc_auto_mode_manager/include/rtc_auto_mode_manager/node.hpp:37]: I fixed at 
[5904eac](https://github.com/autowarefoundation/autoware.universe/pull/2517/commits/5904eac6f159e3f53c61374c87444b7e30841fec)

**Review by rej55 — APPROVED** [review]
LGTM

**Pull request merged**